[AKS] `az aks create/update`: Add new parameter `--enable-ai-toolchain-operator` to enable Kaito addon

helayoty opened the pull request:
**Related command**
az aks create
az aks update

**Description**<!--Mandatory-->
Add `--enable-ai-toolchain-operator` and `--disable-ai-toolchain-operator` to `az aks` cli.

**Testing Guide**
- az aks create <cluster_name> -g <resource_group_name> --enable-ai-toolchain-operator
- az aks update <cluster_name> -g <resource_group_name> --enable-ai-toolchain-operator
- az aks update <cluster_name> -g <resource_group_name> --disable-ai-toolchain-operator

**History Notes**

[Component Name 1] BREAKING CHANGE: `az command a`: Make some customer-facing breaking change
[Component Name 2] `az command b`: Add some customer-facing feature

---

This checklist is used to make sure that common guidelines for a pull request are followed.

- [x] The PR title and description has followed the guideline in [Submitting Pull Requests](https://github.com/Azure/azure-cli/tree/dev/doc/authoring_command_modules#submitting-pull-requests).

- [ ] I adhere to the [Command Guidelines](https://github.com/Azure/azure-cli/blob/dev/doc/command_guidelines.md).

- [ ] I adhere to the [Error Handling Guidelines](https://github.com/Azure/azure-cli/blob/dev/doc/error_handling_guidelines.md).

yonzhan commented:
Thank you for your contribution! We will review the pull request and get back to you soon.

FumingZhang commented:
Depends on SDK generated based on 2025-05-01

FumingZhang commented:
updated milestone to https://github.com/Azure/azure-cli/milestone/157

yanzhudd commented:
Hi @FumingZhang
is this PR ready to review and merge?

yanzhudd commented:
/azp run

yanzhudd commented:
/azp run

FumingZhang commented:
> Depends on SDK generated based on 2025-05-01

the sdk is not ready yet, moving to next milestone 
https://github.com/Azure/azure-cli/milestone/158

helayoty commented:
/azp run

FumingZhang commented on src/azure-cli/azure/cli/command_modules/acs/_params.py:
```diff
@@ -434,6 +434,7 @@ def load_arguments(self, _):
         c.argument('rotation_poll_interval')
         c.argument('enable_sgxquotehelper', action='store_true')
         c.argument('enable_app_routing', action="store_true")
+        c.argument('enable_ai_toolchain_operator', is_preview=True, action='store_true')
```
As the option is now added to official azure-cli, it should not go under preview anymore?

helayoty replied:
Addressed.

FumingZhang commented on src/azure-cli/azure/cli/command_modules/acs/_params.py:
```diff
@@ -633,6 +634,8 @@ def load_arguments(self, _):
         # addons
         c.argument('enable_secret_rotation', action='store_true')
         c.argument('disable_secret_rotation', action='store_true', validator=validate_keyvault_secrets_provider_disable_and_enable_parameters)
+        c.argument('enable_ai_toolchain_operator', is_preview=True, action='store_true')
+        c.argument('disable_ai_toolchain_operator', is_preview=True, action='store_true')
```
same as above

FumingZhang reviewed (commented):
lgtm

helayoty commented:
Queued [live test](https://dev.azure.com/msazure/CloudNativeCompute/_build/results?buildId=130892921&view=results) to validate the change after addressing the feedback rebasing.

helayoty commented:
> Newly added test cases failed in replay mode https://dev.azure.com/azclitools/public/_build/results?buildId=261975&view=logs&j=c4905e22-e505-5bfe-aa0a-49d3f9325fa2&t=cf247c3b-aad8-52b9-12ca-ceb3e5942fcf&l=4222
> 
> > FAILED src/azure-cli/azure/cli/command_modules/acs/tests/latest/test_aks_commands.py::AzureKubernetesServiceScenarioTest::test_aks_create_with_enable_ai_toolchain_operator
> > FAILED src/azure-cli/azure/cli/command_modules/acs/tests/latest/test_aks_commands.py::AzureKubernetesServiceScenarioTest::test_aks_update_with_enable_ai_toolchain_operator
> 
> Newly queued [live test](https://dev.azure.com/msazure/CloudNativeCompute/_build/results?buildId=131029847&view=results) will help generate the correct recording files you'll need. Please download them from pipeline artifacts and commit them accordingly.

Addressed and replaced. However I didn't see the failure in the [latest](https://dev.azure.com/msazure/CloudNativeCompute/_build/results?buildId=131029847&view=logs&j=b162b355-d59d-5864-ce0f-0a70f12dd28b&t=dc59ccd1-231f-538b-777f-33a592c7ca57&l=2833) running pipeline you mentioned.
